n2v/methods/rmks.py

```diff
@@ -6,7 +6,521 @@
+
+            verror = np.linalg.norm(vxc - vxc_old) / self.nbf ** 2
+            if verror < v_tol:
+                print("vxc stops updating.")
+                break
+
+            Derror = np.linalg.norm(self.Da - Da_old) / self.nbf ** 2
+            eerror = (np.linalg.norm(self.eigvecs_a[:Nalpha] - eig_old) / Nalpha)
+            if (Derror < D_tol) and (eerror < eig_tol):
+                print("KSDFT stops updating.")
+                break
+
+            # linear Mixture
+            if mRKS_step != 1:
+                vxc = vxc * (1 - frac_old) + vxc_old * frac_old
+
+            # Save old data.
+            vxc_old = np.copy(vxc)
+            Da_old = np.copy(self.Da)
+            eig_old = np.copy(self.eigvecs_a[:Nalpha])
+
+            vxc_Fock = self.dft_grid_to_fock(vxc, self.Vpot)
+
+            self._diagonalize_with_potential_mRKS(v=vxc_Fock)
+
+            print("Iter: %i, Density Change: %2.2e, Eigenvalue Change: %2.2e, "
+                  "Potential Change: %2.2e." % (mRKS_step, Derror, eerror, verror))
+
+        if vxc_grid is not None:
+            grid_info = self.grid_to_blocks(vxc_grid)
+            grid_info[-1].set_pointers(self.wfn.Da())
+
+            # A larger atol seems to be necessary for user-defined grid
+            vxchole = self._vxc_hole_quadrature(grid_info=grid_info, atol=5e-4)
+            if self.wfn.name() == "CIWavefunction":
+                ebarWF = self._average_local_orbital_energy(self.nt[0],
+                                                            C_a_GFM, eigs_a_GFM, grid_info=grid_info)
+                taup_rho_WF = self._pauli_kinetic_energy_density(self.nt[0],
+                                                                 C_a_NO, eigs_a_NO, grid_info=grid_info)
+            elif self.wfn.name() == "RHF":
+                ebarWF = self._average_local_orbital_energy(self.nt[0],
+                                                            self.ct[0],
+                                                            epsilon_a[:Nalpha],
+                                                            grid_info=grid_info)
+                taup_rho_WF = self._pauli_kinetic_energy_density(self.nt[0],
+                                                                 self.ct[0],
+                                                                 grid_info=grid_info)
+            ebarKS = self._average_local_orbital_energy(self.Da, self.Coca,
+                                                        self.eigvecs_a[:Nalpha], grid_info=grid_info)
+            taup_rho_KS = self._pauli_kinetic_energy_density(self.Da, self.Coca,
+                                                             grid_info=grid_info)
+
+            potential_shift = np.max(ebarWF) - np.max(ebarKS)
+            self.shift = potential_shift
+
+            vxc = vxchole + ebarKS - ebarWF + taup_rho_WF - taup_rho_KS + potential_shift
+
+        return vxc, vxchole, ebarKS, ebarWF, taup_rho_WF, taup_rho_KS
```

Comment:
Can we store this in a data class?
Just trying to generalize the API, since All the other invert methods don't return anything.

Reply:
Saving those as attributes of the class? Or creating a class to save all the 6 components?

Reply:
Anything will do the job. 
I think the cleanest way would be to group them in a data class like in inverter.py

Them you could do something like:

Results = instance_of_dataclass

Self.results.vxc = vxc
Self.results.e_bar = ebar_wfn
.
.
.

Reply:
Good point. Actually the other components exist originally for debugging and I think I will eliminate them and update self.vxc.

Reply:
I currently add the result to self.grid.vxc. But probably we should also have a dataclass for vxc because now in different methods vxc are stored in different ways (WY as v_opt, mRKS in grid.vxc).

Reply:
I opened a new issue so we can discuss the details over there.